pymeasure/instruments/tdk/tdk_base.py

```diff
@@ -0,0 +1,387 @@
+        0 to 255.
+        """,
+        validator=strict_range,
+        values=[0, 255],
+        cast=int
+    )
+
+    over_voltage = Instrument.control(
+        "OVP?", "OVP %g",
+        """Control the over voltage protection.
+        """,
+        validator=lambda v, vs: strict_discrete_range(v, vs, step=0.01),
+        values=[2, 44],
+        dynamic=True
+    )
+
+    under_voltage = Instrument.control(
+        "UVL?", "UVL %g",
+        """Control the under voltage limit.
+
+        Property is UNTESTED.
+        """,
+        validator=lambda v, vs: strict_discrete_range(v, vs, step=0.01),
+        values=[0, 38],
+        dynamic=True
+    )
+
+    auto_restart = Instrument.control(
+        "AST?", "AST %s",
+        """Control the auto restart mode.
+
+        Valid values are ``'ON'`` and ``'OFF'``.
+        """,
+        validator=strict_discrete_set,
+        values=["ON", "OFF"]
+    )
+
+    # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    # Methods
+    # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+    def clear(self):
+        """Clear FEVE and SEVE registers to zero."""
+        self.write("CLS")
+
+    def reset(self):
+        """Reset the instrument to default values."""
+        self.write("RST")
+
+    def foldback_reset(self):
+        """Reset the fold back delay to 0 s, restoring the standard 250 ms
+        delay.
+
+        Property is UNTESTED.
+        """
+        self.write("FDBRST")
+
+    def save(self):
+        """Save current instrument settings."""
+        self.write("SAV")
+
+    def recall(self):
+        """Recall last saved instrument settings."""
+        self.write("RCL")
+
+    def set_max_over_voltage(self):
+        """Set the over voltage protection to the maximum level for the power
+        supply.
+        """
+        self.write("OVM")
+
+    def ramp_to_current(self, target_current, steps=20, pause=0.2):
+        """Ramps to a target current from the set current value over
+        a certain number of linear steps, each separated by a pause duration.
+
+        :param target_current: Target current in amps
+        :param steps: Integer number of steps
+        :param pause: Pause duration in seconds to wait between steps
+        """
+
+        currents = [round(i, 2) for i in np.linspace(self.current,
+                                                     target_current, steps)]
+        for current in currents:
+            self.current = current
+            sleep(pause)
+
+    def shutdown(self):
+        """Safety shutdown the power supply.
+
+        Ramps the power supply down to zero current using the
+        ``self.ramp_to_current(0.0)`` method and turns the output off.
+        """
+        log.info("Shutting down %s." % self.name)
+        self.ramp_to_current(0.0)
+        self.source_output = "OFF"
+        log.info("%s has been shut down." % self.name)
```

Comment:
I think, that this is already done in the `Instrument`'s `shutdown` method.

Reply:
Updated in last commit